nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -398,33 +428,49 @@ def _setup_dataloader_from_config(self, cfg: DictConfig):
             if cfg.get("metadata_file") is None:
                 raise FileNotFoundError("Trying to use tarred data set but could not find metadata path in config.")
             else:
-                metadata_file = cfg.get('metadata_file')
-                with open(metadata_file) as metadata_reader:
-                    metadata = json.load(metadata_reader)
-                if cfg.get('tar_files') is None:
-                    tar_files = metadata.get('tar_files')
-                    if tar_files is not None:
-                        logging.info(f'Loading from tarred dataset {tar_files}')
+                if not self.multilingual:
+                    metadata_file_list = [cfg.get('metadata_file')]
+                else:
+                    metadata_file_list = cfg.get('metadata_file')
+
+                datasets = []
+                for idx, metadata_file in enumerate(metadata_file_list):
+                    with open(metadata_file) as metadata_reader:
+                        metadata = json.load(metadata_reader)
+                    if cfg.get('tar_files') is None:
+                        tar_files = metadata.get('tar_files')
+                        if tar_files is not None:
+                            logging.info(f'Loading from tarred dataset {tar_files}')
+                        else:
+                            raise FileNotFoundError("Could not find tarred dataset in config or metadata.")
                     else:
-                        raise FileNotFoundError("Could not find tarred dataset in config or metadata.")
+                        tar_files = cfg.get('tar_files')
+                        if multilingual:
+                            tar_files = tar_files[idx]
+                        if metadata.get('tar_files') is not None:
+                            logging.info(
+                                f'Tar file paths found in both cfg and metadata using one in cfg by default - {tar_files}'
+                            )
+
+                    dataset = TarredTranslationDataset(
+                        text_tar_filepaths=tar_files,
+                        metadata_path=metadata_file,
+                        encoder_tokenizer=self.encoder_tokenizer,
+                        decoder_tokenizer=self.decoder_tokenizer,
+                        shuffle_n=cfg.get("tar_shuffle_n", 100),
+                        shard_strategy=cfg.get("shard_strategy", "scatter"),
+                        global_rank=self.global_rank,
+                        world_size=self.world_size,
+                        reverse_lang_direction=cfg.get("reverse_lang_direction", False),
+                        prepend_id=self.multilingual_ids[idx],
+                    )
+                    datasets.append(dataset)
+
+                if self.multilingual:
+                    dataset = ConcatTranslationDataset(datasets)
```

Comment:
You probably need to pass other args to `ConcatTranslationDataset` from the config like sampling strategy, temperature values etc right?

Reply:
Added these in the config